stripe/src/main/java/com/stripe/android/FingerprintDataRepository.kt

```diff
@@ -20,25 +21,28 @@ internal interface FingerprintDataRepository {
         private val fingerprintRequestFactory: FingerprintRequestFactory,
         private val fingerprintRequestExecutor: FingerprintRequestExecutor =
             FingerprintRequestExecutor.Default(),
-        private val coroutineScope: CoroutineScope = CoroutineScope(Dispatchers.Main + SupervisorJob())
+        dispatcher: CoroutineDispatcher = Dispatchers.IO
     ) : FingerprintDataRepository {
         private var cachedFingerprintData: FingerprintData? = null
 
         private val timestampSupplier: () -> Long = {
             Calendar.getInstance().timeInMillis
         }
 
+        private val scope = CoroutineScope(dispatcher)
```

Comment:
I'm curious about this change from a [supervised](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-supervisor-job.html) scope to unsupervised. Would you mind explaining here or, better yet, in a comment?

Reply:
Good question. I think my original usage of `SupervisorJob` was unnecessary, because there are no child jobs. Does that make sense?

Reply:
Yep, that makes sense to me.